database.py

```diff
@@ -567,6 +567,26 @@ def toggle_all_bill_shares(self, bill_id, user_id):
             self.is_error = True
             raise e
 
+    def has_bill_share(self, bill_id, item_id, user_id):
+        try:
+            self.cursor.execute("""\
+                SELECT id from bill_shares
+                    WHERE bill_id =  %s
+                        AND item_id = %s
+                        AND user_id = %s
+                        AND is_deleted = FALSE;
+            """, (bill_id, item_id, user_id)
+            )

```

Comment:
One solution to race condition is to use [SQL row locks](https://www.postgresql.org/docs/9.5/static/sql-select.html#SQL-FOR-UPDATE-SHARE). Good example of this shown [here](http://blog.2ndquadrant.com/postgresql-anti-patterns-read-modify-write-cycles/).